### PR TITLE
Add airbnb lint rules

### LINKED ISSUE #144: TS client: use airbnb-typescript eslint config
This will bring the style of this codebase more in line with what the rest of the team uses

https://www.npmjs.com/package/eslint-config-airbnb-typescript

### PR DESCRIPTION
**Pull request by zachmullen**
Fixes #144

### CODE REVIEW

**Comment by zachmullen** [comment]
Depends on #164 , will rebase after that is merged.

**Comment by brianhelba** [comment]
Great improvement. I'll review and merge this on top of #176 , within the next day or so.

**Comment by brianhelba** [comment]
I've rebased this again.

File: javascript-client/package.json
Comment (brianhelba) [inline javascript-client/package.json:50]:
```suggestion
```

`@typescript-eslint/no-non-null-assertion` doesn't currently have any violations in the codebase.

I think it is a good practice to encourage developers to avoid `!.` whenever possible, per the [rationale of the rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-non-null-assertion.md). In some cases, it may indeed be unavoidable, but we can ignore those on a line-by-line basis.

File: javascript-client/package.json
Comment (brianhelba) [inline javascript-client/package.json:51]:
```suggestion
```

I think this is a good rule in general to encourage.

To the extent that we knowingly break it in one place in this code, it's due to very specific rationale around upload bandwidth saturation, not unsoundness in the rule itself. I think it's best to just ignore the single line in this case, rather than globally waive the rule.

File: javascript-client/src/client.ts
Comment (brianhelba) [inline javascript-client/src/client.ts:84]:
```suggestion
      // eslint-disable-next-line no-await-in-loop
      const response = await axios.put(part.upload_url, chunk, {
```

If we don't set this globally, it needs to be ignored here.

**Review by brianhelba — COMMENTED** [review]
Some comments about specific rules inline.

This is a nice improvement and is ready to merge once feedback is addressed.

File: javascript-client/package.json
Comment (zachmullen) [reply javascript-client/package.json]: Sure thing, I didn't add this, but I'll remove it.

File: javascript-client/package.json
Comment (zachmullen) [reply javascript-client/package.json:50]: I've had to disable this enough times now that I've come to just ignore it globally. My attitude is, if I'm using await in a loop, I'm doing it on purpose. I know how to run things in parallel vs. serial, and I think anyone developing on our projects should know that too. This strikes me as a rule to catch more novice developers who don't know about Promise.all.

But I defer to you here. If you find value in this rule rather than just annoyance, let's leave it on.

Comment (brianhelba) [reply javascript-client/package.json]: Got it. That seems like a reasonable thing to watch out for as the codebase continues to mature.

**Review by brianhelba — APPROVED** [review]

**Pull request merged**
